### PR TITLE
pid1: set Description even for devices which don't exist yet

### PR DESCRIPTION
**Pull request by keszybz**
We'd only set the description after the device appeared in sysfs, so
we'd always print
"A start job is running for dev-disk-by\x2duuid-aaaa ... aaaa.device (42s / 1min 30s)"
Let's make this
"A start job is running for /dev/disk/by-duuid/aaaa ... aaaa (42s / 1min 30s)"

https://bugzilla.redhat.com/show_bug.cgi?id=1655860

### CODE REVIEW

**Comment by keszybz** [comment]
The first version didn't actually work as expected, sorry for that. Should be OK now.

File: src/core/device.c [hunk 1]
```diff
@@ -111,6 +111,28 @@ static void device_done(Unit *u) {
...
+        if (!u->description) {
+                /* Generate a description based on the path, to be used until the
+                   device is initialized properly */
+                char *p;
+
+                r = unit_name_to_path(u->id, &p);
+                if (r < 0)
+                        return log_unit_debug_errno(u, r, "Failed to unescape name: %m");
+
+                u->description = p;
```
Comment (poettering) [inline src/core/device.c]: why define `p` here? you can just tell unit_name_to_path() to write directly to `u->description`

Comment (keszybz) [reply src/core/device.c]: Indeed. Updated.

**Comment by evverx** [comment]
As far as I can see, CentOS CI failed because it failed to reboot the machine after it installed `systemd` there. Currently CentOS CI doesn't take the master branch into account (and [I hope it will be changed](https://github.com/systemd/systemd/pull/11052#issuecomment-444362541)) so it could have been caused by the udev bug that was fixed recently. @keszybz could you rebase the branch?

**Comment by evverx** [comment]
@keszybz thank you! To judge https://ci.centos.org/job/systemd-pr-build/3703/console, the issue is gone and the tests are likely to pass.

**Comment by poettering** [comment]
Hmm, with this would mean that if the user drops in a .device unit file for a device we'd override the Description= with the data we generate ourselves, no?

I wonder if it wouldn't be nicer to extend the UnitVTable with a new description verb, then use that in unit_description() if u->description is not set. Then, always generate the description on the go, and no data from a .device unit file was ever read?

**Comment by keszybz** [comment]
> if the user drops in a .device unit file for a device we'd override the Description= with the data we generate ourselves, no?

No, I don't think so. I see
```console
$ systemctl -t device
...
sys-devices-pnp0-00:04-tty-ttyS0.device            loaded active plugged /sys/devices/pnp0/00:04/tty/ttyS0                                 
sys-module-configfs.device                         loaded active plugged my favourite sys-module-configfs                                  
sys-subsystem-net-devices-ens3.device              loaded active plugged Virtio network device
$ systemctl show -p Description /dev/foo sys-module-configfs.device
Description=/dev/foo
FragmentPath=

Description=my favourite sys-module-configfs
FragmentPath=/etc/systemd/system/sys-module-configfs.device
```

**Pull request merged**